### PR TITLE
delete an  offline mutation from cache when it's canceled

### PR DESCRIPTION
**Pull request by sunchunqiang**
*Issue #, if available:*

*Description of changes:*
Fix a bug in AWSPerformMutationOperation.cancle, so that a mutation can be deleted from the persistent database when it's canceled ([Issue #187](https://github.com/awslabs/aws-mobile-appsync-sdk-ios/issues/187))

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by sunchunqiang** [comment]
>     3\. referencing

Done!

**Comment by sunchunqiang** [comment]
>     1. Please add unit or integration tests to replicate this issue, and verify that your change fixes it
> 
>     2. Please update the commit message with a reference to the issue being resolved
> 
>     3. Please add a CHANGELOG entry referencing the issue being resolved

Done!

File: AWSAppSyncUnitTests/MutationQueueTests.swift [hunk 1]
```diff
@@ -616,11 +616,36 @@ class MutationQueueTests: XCTestCase {
 
         wait(for: [mutationWasPerformed], timeout: 1.0)
     }
-
+    
+    func testOfflineMutationCancel() throws {
```
Comment (palpatim) [inline AWSAppSyncUnitTests/MutationQueueTests.swift]: 1. nit: rename to `testCancelingOfflineMutationRemovesFromQueue`
1. Please confirm that this test fails if you comment out your code change.

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ The AWS AppSync SDK for iOS enables you to access your AWS AppSync backend and p
 
 - Fixed incorrect AWSCore dependency version in podspec ([Issue #190](https://github.com/awslabs/aws-mobile-appsync-sdk-ios/issues/190))
 - Fixed data races in AppSyncMQTTClient that were causing crashes ([Issue #184](https://github.com/awslabs/aws-mobile-appsync-sdk-ios/issues/184))
+- Fix a bug in AWSPerformMutationOperation.cancle, so that a mutation can be deleted from the persistent database when it's canceled ([Issue #187](https://github.com/awslabs/aws-mobile-appsync-sdk-ios/issues/187))
```
Comment (palpatim) [inline CHANGELOG.md]: Suggested reword: "Fixed bug where canceling a mutation properly removes it from the persistent mutation cache. ([Issue..."

File: AWSAppSyncUnitTests/MutationQueueTests.swift (hunk 1 above)
Comment (sunchunqiang) [reply AWSAppSyncUnitTests/MutationQueueTests.swift]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (sunchunqiang) [reply CHANGELOG.md]: Done

**Pull request merged**